tests/framework/extensions/clusters/aks/aks_cluster_config.go

```diff
@@ -0,0 +1,93 @@
+	LogAnalyticsWorkspaceName   *string            `json:"logAnalyticsWorkspaceName,omitempty" yaml:"logAnalyticsWorkspaceName,omitempty"`
+	Monitoring                  *bool              `json:"monitoring,omitempty" yaml:"monitoring,omitempty"`
+	NetworkDNSServiceIP         *string            `json:"dnsServiceIp,omitempty" yaml:"dnsServiceIp,omitempty"`
+	NetworkDockerBridgeCIDR     *string            `json:"dockerBridgeCidr,omitempty" yaml:"dockerBridgeCidr,omitempty"`
+	NetworkPlugin               *string            `json:"networkPlugin,omitempty" yaml:"networkPlugin,omitempty"`
+	NetworkPodCIDR              *string            `json:"podCidr,omitempty" yaml:"podCidr,omitempty"`
+	NetworkPolicy               *string            `json:"networkPolicy,omitempty" yaml:"networkPolicy,omitempty"`
+	NetworkServiceCIDR          *string            `json:"serviceCidr,omitempty" yaml:"serviceCidr,omitempty"`
+	NodePools                   *[]AKSNodePool     `json:"nodePools,omitempty" yaml:"nodePools,omitempty"`
+	PrivateCluster              *bool              `json:"privateCluster,omitempty" yaml:"privateCluster,omitempty"`
+	ResourceGroup               string             `json:"resourceGroup,omitempty" yaml:"resourceGroup,omitempty"`
+	ResourceLocation            string             `json:"resourceLocation,omitempty" yaml:"resourceLocation,omitempty"`
+	Subnet                      *string            `json:"subnet,omitempty" yaml:"subnet,omitempty"`
+	Tags                        *map[string]string `json:"tags,omitempty" yaml:"tags,omitempty"`
+	VirtualNetwork              *string            `json:"virtualNetwork,omitempty" yaml:"virtualNetwork,omitempty"`
+	VirtualNetworkResourceGroup *string            `json:"virtualNetworkResourceGroup,omitempty" yaml:"virtualNetworkResourceGroup,omitempty"`
+}
+
+type AKSNodePool struct {
+	AvailabilityZones   *[]string `json:"availabilityZones,omitempty" yaml:"availabilityZones,omitempty"`
+	NodeCount           *int64    `json:"nodeCount,omitempty" yaml:"nodeCount,omitempty"`
+	EnableAutoScaling   *bool     `json:"enableAutoScaling,omitempty" yaml:"enableAutoScaling,omitempty"`
+	MaxPods             *int64    `json:"maxPods,omitempty" yaml:"maxPods,omitempty"`
+	Mode                string    `json:"mode,omitempty" yaml:"mode,omitempty"`
+	Name                *string   `json:"name,omitempty" yaml:"name,omitempty"`
+	OrchestratorVersion *string   `json:"orchestratorVersion,omitempty" yaml:"orchestratorVersion,omitempty"`
+	OsDiskSizeGB        *int64    `json:"osDiskSizeGB,omitempty" yaml:"osDiskSizeGB,omitempty"`
+	OsDiskType          string    `json:"osDiskType,omitempty" yaml:"osDiskType,omitempty"`
+	OsType              string    `json:"osType,omitempty" yaml:"osType,omitempty"`
+	VMSize              string    `json:"vmSize,omitempty" yaml:"vmSize,omitempty"`
+}
+
+func aksNodePoolConstructor(aksNodePoolConfigs *[]AKSNodePool, kubernetesVersion string) *[]management.AKSNodePool {
```

Comment:
nit: in golang constructors are named NewAKSNodePool

Reply:
Yeah I see, in this instance it's my mistake I'm overloading the term constructor. What about `aksNodePoolBuilder` instead? This is a helper function to create an AKSNodePool list from what is provided in the config file.